### PR TITLE
T22051 doc: add staging.md

### PR DESCRIPTION
**Pull request by gctucker**
Add staging.md page with a description of how staging.kernelci.org
works and the development workflow around it.

Signed-off-by: Guillaume Tucker <guillaume.tucker@collabora.com>

### CODE REVIEW

File: doc/staging.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+tool is run.
+
+Things to note:
+
+* Pull requests are only merged from users that are on a trusted list, stored
+  in the `kernelci-deploy` configuration files.
+* Pull requests are merged in order of their number, so older ones take
+  precedence.
+* Pull requests that fail to merge are ignored and will not be tested.
+* If any of the patches stored in `kernelci-deploy` doesn't apply, the
```
Comment (patersonc) [inline doc/staging.md:36]: Perhaps...
```suggestion
* If any of the patches stored in `kernelci-deploy` do not apply, the
```
But I'm not an expert ;)

File: doc/staging.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+A special feature of the staging instance is the ability to test code from
+pending GitHub pull requests before they get merged.  This is handled by tools
+in the [`kernelci-deploy`](https://github.com/kernelci/kernelci-deploy)
+project, to pull all the open pull requests for a given project, apply some
+arbitrary patches and push the resulting `staging.kernelci.org` branch back to
+the repository.  This branch is being replaced (force-pushed) every time the
+tool is run.
+
+Things to note:
```
Comment (patersonc) [inline doc/staging.md:29]: Is it worth adding a note about the "staging-skip" label?

File: doc/staging.md [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+1. update the `kernelci-frontend` service using Ansible from [`kernelci-frontend-config`](https://github.com/kernelci/kernelci-frontend-config) with the staging branch
+1. create and push a `staging.kernelci.org` branch with a tag to the [kernelci Linux kernel repo](https://github.com/kernelci/linux)
+1. trigger a monitor job in Jenkins with the [`kernelci_staging`](https://github.com/kernelci/kernelci-core/blob/staging.kernelci.org/build-configs.yaml#L612) config
+
+The last step should cause the monitor job to detect that the staging kernel
+branch has been updated, and run a kernel build trigger job which in turn will
+cause tests to be run.  Builds and test results will be sent to the staging
+backend instance, and results will be available on the staging web dashboard.
+Regressions will cause bisections to be run on the staging instance, and
+results to be sent to the `kernelci-results-staging@groups.io` mailing list.
```
Comment (patersonc) [inline doc/staging.md:75]: You could link the email address here to the groups.io page so people know where to sign up.

File: doc/staging.md (hunk 2 above)
Comment (gctucker) [reply doc/staging.md:29]: Good point, I'll add that.  Thanks.

File: doc/staging.md (hunk 1 above)
Comment (gctucker) [reply doc/staging.md:36]: Not 100% sure, but I believe it's singular as in "if any applies" -> "if any doesn't apply" -> "if any of them doesn't apply".

Comment (gctucker) [reply doc/staging.md]: Fixed using "if any patch", so it's all singular now :)

File: doc/staging.md (hunk 2 above)
Comment (gctucker) [reply doc/staging.md:29]: Done, added a bullet point a bit further down.

File: doc/staging.md (hunk 3 above)
Comment (gctucker) [reply doc/staging.md]: Done.

File: doc/staging.md (hunk 2 above)
Comment (patersonc) [reply doc/staging.md:29]: LGTM

File: doc/staging.md (hunk 3 above)
Comment (patersonc) [reply doc/staging.md]: LGTM

**Pull request merged**